### ISSUE TITLE
Shell extensions

### ISSUE THREAD

**Issue by qzed** [linux-surface/surface-dtx-daemon#6]
The per-user deamon could be replaced entirely with a Gnome shell extension (or something similar for KDE/other DEs). This would also allow requesting detachment via UI and may be be a bit more flexible with regards to locking/unlocking the latch (e.g. like listing applications that use the dGPU and need to be closed before detaching), once we've figured out how to handle that (see #5). For now simply adding a sort of request button and handling notifications should be enough. 

Everything required for that should be already provided in the D-Bus service, so this extension just needs to communicate with that.

**Comment by StollD** [comment]
An alternative to a full blown extension might be to implement support for appindicators into the per-user daemon. That would allow for UI integration in a cross-DE way. It would still need an extension on GNOME though but that already exists and seems to be pretty well maintained.

**Comment by qzed** [comment]
Yeah, I think that'd be an acceptable alternative. I don't really plan on doing this myself any time soon (which is kinda why I opened this issue), but I'm happy to accept PRs towards this.
